packages/dds/map/src/mapKernel.ts

```diff
@@ -548,6 +548,18 @@ export class MapKernel {
         return previousLocalValue;
     }
 
+    private getClearLocalOpMetadata(op: IMapClearOperation, previousMap?: Map<string, ILocalValue>): unknown {
+        return { type: "clear", pendingMessageId: this.getMapClearMessageId(), previousMap };
+    }
+
+    private getKeyLocalOpMetadata(op: IMapKeyOperation, previousValue?: ILocalValue): unknown {
```

Comment:
the pattern of getting the metadata after the fact is strange to me, but i also don't know this code super well so maybe it is necessary. it seems to me a better pattern would be to construct  the metadata through the process of doing the local operation, so doing a local clear creates a local op metadata for clear, which is gets passed up the stack

Reply:
ok. it looks like we previously create it right before submit, so i guess this pattern makes sense, but i could change a couple things. i would move both the functions outside the class, and i would name then create rather than get, so 

`function createKeyLocalOpMetadata(...)`

Reply:
I agree with this but part of the metadata "pendingMessageId" is making a call to get the message id which will not just get it but increment the counter, which is also super weird to have while just trying to get the metadata. The pendingmessageid should be already defined. I added another task which involves a refactoring to mapKernel.ts https://dev.azure.com/fluidframework/internal/_workitems/edit/1272/ but I believe this PR goes to the right direction.

Reply:
the function can still just take that value, and the caller gets it. i'd rather not nest the bad patterns deeper which makes it even harder to understand.

Reply:
it keeps these functions very simple, in that they just construct the object

Reply:
I believe we can't take them outside the class since they call this.getMapClearMessageId() which increments this.pendingMessageId

Reply:
you can. you just have the caller call this.getMapClearMessageId() and pass that as a parameter to the create functions